docker/creds/credentials_test.go

```diff
@@ -643,20 +644,33 @@ func correctVerifyCbk(ctx context.Context, image string, credentials Credentials
 	return creds.ErrUnauthorized
 }
 
-func withCleanHome(t *testing.T) {
-	t.Helper()
-	homeName := "HOME"
+func testHomeEnvName(t *testing.T) string {
 	if runtime.GOOS == "windows" {
-		homeName = "USERPROFILE"
+		return "USERPROFILE"
 	}
+	return "HOME"
+}
+
+func withCleanHome(t *testing.T) {
+	t.Helper()
 	tmpHome := t.TempDir()
-	t.Setenv(homeName, tmpHome)
+	t.Setenv(testHomeEnvName(t), tmpHome)
 
 	if runtime.GOOS == "linux" {
 		t.Setenv("XDG_CONFIG_HOME", filepath.Join(tmpHome, ".config"))
 	}
 }
 
+func testConfigPath(t *testing.T) string {
```

Comment:
I believe this should return the same value as `config.Path()` most of the times, doesn't it?

Reply:
Hmm actually not, because `homedir` pkg is caching. Why do we use that package and not just `os.UserHomeDir()`.

Reply:
Or we could disable caching in tests, `homedir.DisableCache = true`.